homeassistant/components/home_connect/switch.py

```diff
@@ -383,46 +366,53 @@ async def async_turn_off(self, **kwargs: Any) -> None:
                 translation_key="power_off",
                 translation_placeholders={
                     **get_dict_from_home_connect_error(err),
-                    SVE_TRANSLATION_PLACEHOLDER_APPLIANCE_NAME: self.device.appliance.name,
+                    SVE_TRANSLATION_PLACEHOLDER_APPLIANCE_NAME: self.appliance.info.name,
                     SVE_TRANSLATION_PLACEHOLDER_VALUE: self.power_off_state,
                 },
             ) from err
-        self.async_entity_update()
 
-    async def async_update(self) -> None:
-        """Update the switch's status."""
-        if (
-            self.device.appliance.status.get(BSH_POWER_STATE, {}).get(ATTR_VALUE)
-            == BSH_POWER_ON
-        ):
+    def update_native_value(self) -> None:
+        """Set the value of the entity."""
+        power_state = self.appliance.settings[SettingKey.BSH_COMMON_POWER_STATE]
+        value = cast(str, power_state.value)
+        if value == BSH_POWER_ON:
             self._attr_is_on = True
         elif (
             hasattr(self, "power_off_state")
-            and self.device.appliance.status.get(BSH_POWER_STATE, {}).get(ATTR_VALUE)
-            == self.power_off_state
+            and self.power_off_state
+            and value == self.power_off_state
         ):
             self._attr_is_on = False
+        elif not hasattr(self, "power_off_state") and value in [
+            BSH_POWER_OFF,
+            BSH_POWER_STANDBY,
+        ]:
+            self.power_off_state = value
+            self._attr_is_on = False
         else:
             self._attr_is_on = None
-        _LOGGER.debug("Updated, new state: %s", self._attr_is_on)
 
     async def async_fetch_power_off_state(self) -> None:
         """Fetch the power off state."""
-        try:
-            data = await self.hass.async_add_executor_job(
-                self.device.appliance.get, f"/settings/{self.bsh_key}"
-            )
-        except HomeConnectError as err:
-            _LOGGER.error("An error occurred: %s", err)
-            return
-        if not data or not (
-            allowed_values := data.get(ATTR_CONSTRAINTS, {}).get(ATTR_ALLOWED_VALUES)
-        ):
+        data = self.appliance.settings.get(
+            SettingKey.BSH_COMMON_POWER_STATE,
+        )
+        if not data or not data.constraints or not data.constraints.allowed_values:
+            try:
+                data = await self.coordinator.client.get_setting(
```

Comment:
I don't understand why we need to fetch this explicitly here. We seem to be asking for the data for the `SettingKey.BSH_COMMON_POWER_STATE` which is the same key used for the state of this entity. This entity isn't added unless that key exist in the coordinator state. We have that check above in `async_setup_entry`. Why wouldn't all data be available before creating the entity if the key is there?

Reply:
I'd fetch the settings before creating and adding the entity so we don't need to do that later.

Reply:
We don't need to change this in this PR though.

Reply:
I think it is better where it is already, because fetching it here allows us to fetch it only if the power setting is present in the appliance (this can be also done at wherever we would move it) and if the entity is enabled.

We could even move it to the turn off method so that we fetch it only when it is strictly necessary (although it might consume some "user time" instead of doing it while the user doesn't explicitly need it)